lib/Onyx.d.ts

```diff
@@ -0,0 +1,337 @@
+type ConnectOptions<TKey extends OnyxKey> = BaseConnectOptions<TKey> &
+    (
+        | {
+              key: TKey extends CollectionKey ? TKey : never;
+              callback?: (value: OnyxCollectionRecords<KeyValueMapping[TKey]>, key?: TKey) => void;
+              waitForCollectionCallback: true;
+          }
+        | {
+              key: TKey;
+              callback?: (value: OnyxRecord<KeyValueMapping[TKey]>, key?: TKey) => void;
+              waitForCollectionCallback?: false;
+          }
+    );
+
+/**
+ * Represents a mapping between Onyx collection keys and their respective values.
+ * It helps to enforce that a Onyx collection key should not be without suffix (e.g. should always be of the form `${TKey}${string}`),
+ * and to map each Onyx collection key with suffix to a value of type `TValue`.
+ */
+type Collection<TKey extends Key, TMap, TValue> = {
+    [MapK in keyof TMap]: MapK extends `${TKey}${string}`
+        ? MapK extends `${TKey}`
+            ? never // forbids empty id
+            : TValue
+        : never;
+};
+
+/**
+ * Represents different kinds of updates that can be passed to `Onyx.update()` method. It is a discriminated union of
+ * different update methods (`SET`, `MERGE`, `MERGE_COLLECTION`), each with their own key and value structure.
+ */
+type OnyxUpdate<TKey extends OnyxKey> =
+    | {
+          onyxMethod: typeof METHOD.SET;
+          key: TKey;
+          value: OnyxRecord<KeyValueMapping[TKey]>;
+      }
+    | {
+          onyxMethod: typeof METHOD.MERGE;
+          key: TKey;
+          value: PartialDeep<KeyValueMapping[TKey]>;
+      }
+    | {
+          [TKey in CollectionKey]: {
+              onyxMethod: typeof METHOD.MERGE_COLLECTION;
+              key: TKey;
+              value: Record<`${TKey}${string}`, PartialDeep<KeyValueMapping[TKey]>>;
+          };
+      }[CollectionKey];
+
+/**
+ * Represents a record of `OnyxUpdate`s to be passed to `Onyx.update()` method, indexed by keys of type `OnyxKey`.
+ * Each `OnyxUpdate` will be associated with its respective Onyx key, ensuring different type-safety for each object.
+ */
+type OnyxUpdates<TKeyList extends OnyxKey[]> = {
+    [TKey in keyof TKeyList]: OnyxUpdate<TKeyList[TKey]>;
+};
+
+/**
+ * Represents the options used in `Onyx.init()` method.
+ */
+type InitOptions = {
+    keys?: DeepRecord<string, string>;
+    initialKeyStates?: Partial<NullableKeyValueMapping>;
+    safeEvictionKeys?: OnyxKey[];
+    maxCachedKeysCount?: number;
+    captureMetrics?: boolean;
+    shouldSyncMultipleInstances?: boolean;
+    debugSetState?: boolean;
+};
+
+declare const METHOD: {
+    readonly SET: 'set';
+    readonly MERGE: 'merge';
+    readonly MERGE_COLLECTION: 'mergecollection';
+    readonly CLEAR: 'clear';
+};
+
+/**
+ * Returns current key names stored in persisted storage
+ */
+declare function getAllKeys(): Promise<Array<OnyxKey>>;
+
+/**
+ * Checks to see if this key has been flagged as
+ * safe for removal.
+ *
+ * @param testKey
```

Comment:
`@param` without description should be removed, these don't provide much value.

Reply:
@blazejkustra I tried to keep the same comment it is in `Onyx.js`, like the other functions.

Reply:
yes if there is no description for a parameter, then we can remove that parameter from the JSDoc

Reply:
@hayata-suenaga @blazejkustra removed!